clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/client/RegionScopedTitusClient.java

```diff
@@ -0,0 +1,202 @@
+    private static final long DEFAULT_READ_TIMEOUT = 20000;
+
+    /** An instance of {@link TitusRegion} that this RegionScopedTitusClient will use */
+    private final TitusRegion titusRegion;
+
+    /** Retrofit rest adapter for the titus environment */
+    private final TitusRestAdapter titusRestAdapter;
+
+    /** Actual connect timeout that will be used by the retrofit client */
+    private final long connectTimeoutMillis;
+
+    /** Actual read timeout that will be used by the retrofit client */
+    private final long readTimeoutMillis;
+
+    /** Titus client uses Jackson converter for retrofit. Titus client users can override this */
+    private final ObjectMapper objectMapper;
+
+    public RegionScopedTitusClient(TitusRegion titusRegion) {
+        this(titusRegion, DEFAULT_CONNECT_TIMEOUT, DEFAULT_READ_TIMEOUT, TitusClientObjectMapper.configure());
+    }
+
+    public RegionScopedTitusClient(TitusRegion titusRegion,
+                                   long connectTimeoutMillis,
+                                   long readTimeoutMillis,
+                                   ObjectMapper objectMapper) {
+
+        this.titusRegion = titusRegion;
+        this.connectTimeoutMillis = connectTimeoutMillis;
+        this.readTimeoutMillis = readTimeoutMillis;
+        this.objectMapper = objectMapper;
+        this.titusRestAdapter = createTitusRestAdapter(titusRegion);
+    }
+
+    private TitusRestAdapter createTitusRestAdapter(TitusRegion titusRegion) {
+        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor((msg) -> LOGGER.info("[{}] {}", titusRegion.getName(), msg));
+        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+        OkHttpClient okHttpClient = new OkHttpClient.Builder()
+                .connectTimeout(connectTimeoutMillis, TimeUnit.MILLISECONDS)
+                .readTimeout(readTimeoutMillis, TimeUnit.MILLISECONDS)
+                .addNetworkInterceptor(interceptor)
+                .build();
+        return new Retrofit.Builder()
+                .baseUrl(titusRegion.getEndpoint())
+                .client(okHttpClient)
+                .addConverterFactory(JacksonConverterFactory.create(objectMapper))
+                .validateEagerly(true)
+                .build()
+                .create(TitusRestAdapter.class);
+    }
+
+
+    // APIs
+    // ------------------------------------------------------------------------------------------
+
+    @Override
+    public Job getJob(String jobId) {
+        return execute(titusRestAdapter.getJob(jobId));
+    }
+
+    private <T> T execute(Call<T> call) {
+        try {
+            retrofit2.Response<T> response = call.execute();
+            if (response.isSuccess()) {
+                return response.body();
+            }
+            throw new RuntimeException("response failed " + response.code() + " " + response.message());
+        } catch (IOException ioe) {
+            throw new RuntimeException(ioe);
```

Comment:
not exactly - I think for somewhere like orca or `DiscoverySupport` in here where we are really dependent on RetrofitError semantics we will need to build our own equivalent (I think the pieces are available, and honestly we could probably do a safer job if we just shoot for identifying retryable vs network error (e.g. we probably shouldn't retry a POST on a network error unless we are certain that we never actually connected and issued the request))